pkg/informer/informer.go

```diff
@@ -58,14 +59,64 @@ const (
 	byGroupVersionResourceIndex = "byGroupVersionResource"
 )
 
-// DynamicDiscoverySharedInformerFactory is a SharedInformerFactory that
+// GVRPartialMetadata provides the required metadata about a GVR
+// for which an informer should be started.
+type GVRPartialMetadata struct {
+	Names apiextensionsv1.CustomResourceDefinitionNames
+	Scope apiextensionsv1.ResourceScope
+}
+
+// GVRSource is a source of information about available GVRs,
+// and provides a way to register for changes in the available GVRs (addition or removal)
+// so that the list of informers can be updated.
+type GVRSource interface {
+	// GVRs returns the required metadata about all GVRs known by the GVRSource
+	GVRs() map[schema.GroupVersionResource]GVRPartialMetadata
+
+	// Synced returns true if informers used by this GVRSource are synced
+	Synced() bool
```

Comment:
Unclear semantics. As GVRs is giving me potentially incomplete information anyway, what does this func give me?

Reply:
The `GVRSource` is possibly based on an informer (CRD informer, or SyncTarget informer in the case of the Syncer). This `Synced`  method returns true when the GVRSource is ready to return a complete list of supported GVRs (=> typically when the underlying informer is Synced).
Maybe this function should be called `Ready` ?

Reply:
I renamed this `Ready` as proposed.